test/io/test_io.py

```diff
@@ -1214,3 +1214,26 @@ def test_jwt_cache_with_no_exp_claim(defaultenv):
         # their difference should be atleast 300, implying
         # that JWT Caching is working as expected
         assert (first_dur - second_dur) > 300.0
+
+
+def test_cors_allowed_origin_config_no_default(defaultenv):
+    "OPTIONS request should return Access-Control-Allow-Origin with config"
+
+    env = {
+        **defaultenv,
+        "PGRST_CORS_ALLOWED_ORIGINS": "http://example.com, http://example2.com",
+    }
+
+    headers = {
+        "Accept": "*/*",
+        "Origin": "http://example.com",
+        "Access-Control-Allow-Method": "POST",
+        "Access-Control-Allow-Headers": "Content-Type",
+    }
+
+    with run(env=env) as postgrest:
+        response = postgrest.session.options("/items", headers=headers)
+        assert (
+            response.headers["Access-Control-Allow-Origin"]
+            == "http://example.com, http://example2.com"
```

Comment:
This response header [must have only one origin](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Access-Control-Allow-Origin#directives), which is the same as the one in the `Origin` request header (if it's allowed). Having both as it is right now will invalidate the preflight (the `wai-cors` library should handle this nicely, except for `OPTIONS` as I mentioned before).